tools/bqtag/tests/test_bqtag.py

```diff
@@ -0,0 +1,182 @@
+                          "type": "RECORD"
+                        }
+                      ]"""
+
+        TAG_MAP = dict()
+        TAG_MAP["default_column_tag"] = "low"
+        TAG_MAP["column2"] = "medium"
+        TAG_MAP["parent.nested1"] = "high"
+
+
+        bq = BQTableView(bq_dataset = BQ_DATASET,
+                     catalog_taxonomy = TAXONOMY_DISPLAY_NAME,
+                     location = LOCATION,
+                     bq_project = BQ_PROJECT,
+                     catalog_project = CATALOG_PROJECT,
+                     json_credentials_path = JSON_CREDENTIALS_FILE)
+
+        bq.fetch_policy_tags()
+
+        # Create Dataset
+        bq.create_dataset()
+        
+        created_schema = json.loads(bq.create_table(TABLE_TO_CREATE, TABLE_SCHEMA, TAG_MAP))
+
+        # Check if table was created
+        self.assertTrue(created_schema!={}) 
+
+        check = True
+
+        for column in created_schema:
+
+            if column["name"] == "column1":
+                if column["policyTags"]["names"][0] != bq.policy_tags["low"]:
+                    check = False
+                    continue
+
+            if column["name"] == "column2":
+              if column["policyTags"]["names"][0] != bq.policy_tags["medium"]:
+                check = False
+                continue
+          
+            if column["name"] == "parent":
+                for nested_column in column["fields"]:
+                    
+                    if nested_column["name"] == "nested1":
+                        if nested_column["policyTags"]["names"][0] != bq.policy_tags["high"]:
+                            check = False
+                            continue
+
+                    if nested_column["name"] == "nested2":
+                        if nested_column["policyTags"]["names"][0] != bq.policy_tags["low"]:
+                            check = False
+                            continue
+              
+        if len(created_schema) != 3:
+          check = False
+        
+        # Check if tags are correct
+        self.assertTrue(check) 
+            
+
+    def test_4_view_creation(self):
+      """
+      Test view creation
+      """
+      
+      bq = BQTableView(bq_dataset = BQ_DATASET,
+                     catalog_taxonomy = TAXONOMY_DISPLAY_NAME,
+                     location = LOCATION,
+                     bq_project = BQ_PROJECT,
+                     catalog_project = CATALOG_PROJECT,
+                     json_credentials_path = JSON_CREDENTIALS_FILE)
+
+      bq.fetch_policy_tags()
+      
+      SOURCE_TABLE = "test_table"
+      VIEW1 = "test_view_medium"
+      VIEW1_TAGS = ["medium", "low"]
+      
+      query = bq.create_view(SOURCE_TABLE, VIEW1, VIEW1_TAGS)
+
+      self.assertTrue(query.strip() == "SELECT column2, column1, STRUCT(parent.nested2)  as parent FROM `{project}.{dataset}.{table}`".format(project=bq.bq_project, dataset=bq.dataset, table=SOURCE_TABLE))
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Blank line at the end of Python file (PEP8)

Reply:
updated